### PR TITLE
allow public access to GET /api/v1/users/GUID; and include nested FileUpload data

### PR DESCRIPTION
**Pull request by naknomum**
## Pull Request Overview

- Modifies behavior of `GET /api/v1/users/GUID` to allow public access, but based returned content on different Schema depending on privileges
- Provides a **profile_fileupload** value which includes a _nested_ FileUpload schema

### Changes in tests
- Though minimal change in code to provide different output for this GET api endpoint, it affected quite a few tests.  Thus, most changes were made there.

---

**Pull Request Checklist**
- [ ] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [ ] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [ ] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [ ] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [ ] Ensure that the PR is properly covered
  - Example: The percentage of the code covered by tests has not decreased
- [ ] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

File: tests/modules/users/resources/test_modifying_users_info.py [hunk 1]
```diff
@@ -417,7 +420,7 @@ def cleanup():
         }
         response = flask_app_client.patch(*args, **kwargs)
         assert response.status_code == 200, response.data
-        response_asset_guid = response.json['profile_fileupload_guid']
+        response_asset_guid = response.json['profile_fileupload']['guid']
```
Comment (hwindsor) [inline tests/modules/users/resources/test_modifying_users_info.py:423]: and this is why I'm a big fan or write-it-once-use-it-many-times programming a'la the module utils.
Users doesn't have one and I think this proves it needs one

File: app/modules/users/resources.py [hunk 2]
```diff
@@ -107,19 +106,19 @@ class UserByID(Resource):
...
         """
         Get user details by ID.
         """
-        return user
+        from app.modules.users.permissions import rules
+        from app.modules.users.schemas import DetailedUserSchema, PublicUserSchema
+        if not current_user.is_anonymous and (rules.owner_or_privileged(current_user, user) or current_user.is_admin):
+            schema = DetailedUserSchema()
+        else:
+            schema = PublicUserSchema()
```
Comment (hwindsor) [inline app/modules/users/resources.py:120]: This does seem inconsistent in terms of permission. Only admin users can read the list of user GUIDs, but anyone can read their names once they have the guid.
I may have missed a conversation but this seems very strange

**Review by hwindsor — APPROVED** [review]
LGTM,

File: app/modules/users/resources.py (hunk 2 above)
Comment (naknomum) [reply app/modules/users/resources.py:120]: after some slack discussion, i think indeed there will be some addressing we need to do here -- especially making sure what is exposed is consistent with our terms & conditions etc.  basically this is a kind of open area of discussion.

**Pull request merged**